data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/CsaProfileRemedialActionsCreator.java

```diff
@@ -159,10 +158,26 @@ private void addOnContingencyStateUsageRules(RemedialActionAdder<?> remedialActi
                 .withUsageMethod(usageMethod).add());
     }
 
-    private void checkAllContingenciesLinkedToRaHaveTheSameConstraintKind(String remedialActionId, Set<PropertyBag> linkedContingencyWithRAs, String firstKind) {
+    private void checkElementCombinationConstraintKindsCoherence(String remedialActionId, Set<PropertyBag> linkedContingencyWithRAs) {
+        // The same contingency cannot have different Element Combination Constraint Kinds
+        Map<String, Boolean> numberOfIncludedPerContingency = new HashMap<>();
+        Map<String, Boolean> numberOfConsideredPerContingency = new HashMap<>();
         for (PropertyBag propertyBag : linkedContingencyWithRAs) {
-            if (!propertyBag.get(CsaProfileConstants.COMBINATION_CONSTRAINT_KIND).equals(firstKind)) {
-                throw new FaraoImportException(ImportStatus.INCONSISTENCY_IN_DATA, CsaProfileConstants.REMEDIAL_ACTION_MESSAGE + remedialActionId + " will not be imported because ElementCombinationConstraintKind of a remedial action linked to a contingency must be all of the same kind");
+            String combinationKind = propertyBag.get(CsaProfileConstants.COMBINATION_CONSTRAINT_KIND);
+            String contingencyId = propertyBag.get(CsaProfileConstants.REQUEST_CONTINGENCY);
+            if (combinationKind.equals(CsaProfileConstants.ElementCombinationConstraintKind.INCLUDED.toString())) {
+                numberOfIncludedPerContingency.put(contingencyId, true);
+            } else if (combinationKind.equals(CsaProfileConstants.ElementCombinationConstraintKind.CONSIDERED.toString())) {
+                numberOfConsideredPerContingency.put(contingencyId, true);
+            } else if (combinationKind.equals(CsaProfileConstants.ElementCombinationConstraintKind.EXCLUDED.toString())) {
+                throw new FaraoImportException(ImportStatus.INCONSISTENCY_IN_DATA, CsaProfileConstants.REMEDIAL_ACTION_MESSAGE + remedialActionId + " will not be imported because of an illegal EXCLUDED ElementCombinationConstraintKind");
+            } else {
+                throw new FaraoImportException(ImportStatus.INCONSISTENCY_IN_DATA, CsaProfileConstants.REMEDIAL_ACTION_MESSAGE + remedialActionId + " will not be imported because of an invalid Element Combination Constraint Kind");
+            }
+            Boolean hasIncluded = numberOfIncludedPerContingency.get(contingencyId);
+            Boolean hasConsidered = numberOfConsideredPerContingency.get(contingencyId);
+            if ((hasIncluded != null) && (hasConsidered != null) && hasIncluded && hasConsidered) {
```

Comment:
if you have 2 sets, there will be no need of null testing and this test will be : 
if(setContingenciesWithIncluded.contains(contingencyId) && setContingenciesWithConsidered.contains(contingencyId)